### PR TITLE
Rework from extension to plugin

### LINKED ISSUE #9720: [Theia][Plugin]Rework SSH extension to SSH plugin
The existing https://github.com/eclipse/che-theia-ssh-plugin is developed based on the existing extensibility model in Theia IDE. But as far as new plugin model is on it's way and will be used in Che as the main one - there is need to rework SSH extension to SSH plugin.

### PR DESCRIPTION
**Pull request by vinokurig**
Rework the extension to plugin for Theia
fixes https://github.com/eclipse/che/issues/9720

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,73 +1,15 @@
-# Theia - SSH extension
-The example of how to build the Theia-based applications with the SSH extension.
+# Theia - SSH Extension
```
Comment (azatsarynnyy) [inline README.md]: it's not extension anymore but `SSH Plug-in`

**Review by azatsarynnyy — APPROVED** [review]
works well :+1:

File: README.md (hunk 1 above)
Comment (vinokurig) [reply README.md]: fixed

File: src/common/ssh-protocol.ts [hunk 2]
```diff
@@ -35,7 +42,7 @@ export interface SshKeyPair {
      * Che service that uses SSH key pair, e.g. workspace, machine, vcs.
      */
     service: string;
-    /**
+    /**
```
Comment (AndrienkoAleksandr) [inline src/common/ssh-protocol.ts]: I guess it's not needed change.

Comment (vinokurig) [reply src/common/ssh-protocol.ts]: fixed

File: package.json [hunk 3]
```diff
@@ -1,22 +1,44 @@
...
-    "browser-app",
-    "electron-app"
-  ]
+      "name": "che-theia-ssh-plugin",
+      "publisher": "theia",
+      "keywords": [
+        "theia-plugin"
+      ],
+      "version": "0.0.1",
+      "license": "EPL-1.0",
```
Comment (benoitf) [inline package.json]: files should be EPL 2.0 now

File: src/common/ssh-protocol.ts [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
- * which is available at https://www.eclipse.org/legal/epl-2.0/
- *
- * SPDX-License-Identifier: EPL-2.0
- **********************************************************************/
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
 
 export const sshKeyServicePath = '/services/ssh-key';
 export const SshKeyServer = Symbol("SshKeyServer");
```
Comment (benoitf) [inline src/common/ssh-protocol.ts]: AFAIK we don't need anymore Symbols

File: package.json (hunk 3 above)
Comment (vinokurig) [reply package.json]: fixed

File: src/common/ssh-protocol.ts (hunk 4 above)
Comment (vinokurig) [reply src/common/ssh-protocol.ts]: fixed

File: src/che-theia-ssh-plugin-backend-plugin.ts [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import * as theia from '@theia/plugin';
+import { CheService, SshKeyPair } from "./common/ssh-protocol";
```
Comment (benoitf) [inline src/che-theia-ssh-plugin-backend-plugin.ts]: minor issue but in the code sometimes `"` is used, sometimes `'`

**Review by benoitf — APPROVED** [review]
nice to have a plug-in now.

side note: it should be merged only after tagging the current code and changing it there
https://github.com/eclipse/che/blob/master/dockerfiles/theia/src/extensions.json#L14

else after this PR is merged it will fail the theia image

File: src/che-theia-ssh-plugin-backend-plugin.ts (hunk 5 above)
Comment (vinokurig) [reply src/che-theia-ssh-plugin-backend-plugin.ts]: fixed

**Comment by vinokurig** [comment]
I am going to PTO, will follow it in 3 weeks

**Review by vparfonov — APPROVED** [review]

**Pull request merged**